src/linter/root.go

```diff
@@ -484,6 +484,76 @@ func (d *RootWalker) Report(n node.Node, level int, checkName, msg string, args
 	}
 }
 
+// ReportByLine registers a single report message about some found problem in lineIndex code line.
+func (d *RootWalker) ReportByLine(lineIndex int, level int, checkName, msg string, args ...interface{}) {
```

Comment:
Can we make a generalized report method that can implement major logic so we don't duplicate the code that manipulates that hash computations, etc?